util/topgen/clocks.py

```diff
@@ -160,3 +160,15 @@ def add_clock_to_group(self, grp: Group, clk_name: str, src_name: str):
                              f'{grp.name}: the given source name is '
                              f'{src_name}, which is unknown.')
         grp.add_clock(clk_name, src)
+
+    def get_clock_by_name(self, name: str) -> object:
+
+        ret = self.srcs.get(name, None)
+
+        if not ret:
+            ret = self.derived_srcs.get(name, None)
+
+        if ret:
+            return ret
+        else:
+            raise ValueError(f'{name} is not a valid clock')
```

Comment:
You can use `self.all_srcs` here to make this a bit simpler. Also, the Python could be written more simply as:
```python3
    def get_clock_by_name(self, name: str) -> object:
        ret = self.all_srcs.get(name)
        if ret is None:
            raise ValueError(f'{name} is not a valid clock')
        return ret
```
However, I'd probably not define this function at all, and just do the dictionary lookup in the `Resets` constructor.

Reply:
is there any downside to having the function? I added it mostly if things changed in the future, i don't need to modify anything outside of `clocks.py`, what do you think?

Reply:
Happy either way - just being lazy and deleting as much code as possible :-)